components/centraldashboard/public/components/registration-page_test.js

```diff
@@ -59,9 +63,41 @@ describe('Registration Page', () => {
         input.fireEnter();
 
         $e('.Main-Content .iron-selected .actions > paper-button:nth-of-type(1)').click();
+        await sleep(5);
 
+        expect(flowcomplete).toHaveBeenCalled();
+    });
+
+    it('Should work when API DB is not consistent (use polling)', async () => {
+        mockIronAjax(
+            registrationPage.$.GetMyNamespace,
+            {hasWorkgroup: false, hasAuth: true, user: 'test'},
+        );
+        mockIronAjax(
+            registrationPage.$.MakeNamespace,
+            {message: 'Success!'},
+        );
+
+        const $e = (selector) => registrationPage.shadowRoot.querySelector(selector);
+        flush();
         await yieldForAsync(); // So the view can render
 
+        $e('.Main-Content .iron-selected .actions > paper-button').click();
+        const input = registrationPage.$.Namespace;
+        input.value = 'kubeflow';
+        input.fireEnter();
+
+        $e('.Main-Content .iron-selected .actions > paper-button:nth-of-type(1)').click();
+
+        await sleep(5);
+        expect(flowcomplete).not.toHaveBeenCalled();
+
+        mockIronAjax(
+            registrationPage.$.GetMyNamespace,
+            {hasWorkgroup: true, hasAuth: true, user: 'test'},
+        );
+
+        await sleep(300);
```

Comment:
If you want to avoid having real waits in your tests, you can use the Jasmine fake clock to mock out the timer behavior and advance it to run any timers.

https://jasmine.github.io/api/3.5/Clock.html

Reply:
It's not about the timer, as it is about scheduling, does clock control tick scheduling?

Reply:
Okay seems like it works, but it is really nuanced (updated tests to use that!)